src/fees/saga.test.ts

```diff
@@ -245,4 +247,57 @@ describe(estimateFeeSaga, () => {
       )
       .run()
   })
+
+  it('marks as error and shows banner if fee estimation fails', async () => {
+    await expectSaga(estimateFeeSaga, {
+      payload: { feeType: FeeType.SEND, tokenAddress: mockCusdAddress },
+    })
+      .withState(store.getState())
+      .provide([
+        [matchers.call.fn(buildSendTx), jest.fn(() => ({ txo: mockTxo }))],
+        [matchers.call.fn(estimateGas), throwError(new Error('cannot estimate fee'))],
+      ])
+      .put(
+        feeEstimated({
+          feeType: FeeType.SEND,
+          tokenAddress: mockCusdAddress,
+          estimation: {
+            loading: false,
+            error: true,
+            usdFee: null,
+            lastUpdated: Date.now(),
+          },
+        })
+      )
+      .put(
+        showErrorOrFallback(new Error('cannot estimate fee'), ErrorMessages.CALCULATE_FEE_FAILED)
+      )
+      .run()
+  })
+
+  it('marks as error without fee banner if token balance is zero', async () => {
+    const { effects } = await expectSaga(estimateFeeSaga, {
+      payload: { feeType: FeeType.SEND, tokenAddress: mockCeurAddress },
+    })
+      .withState(store.getState())
+      .provide([
+        [matchers.call.fn(buildSendTx), jest.fn(() => ({ txo: mockTxo }))],
+        [matchers.call.fn(estimateGas), throwError(new Error('cannot estimate fee'))],
```

Comment:
We don't need to provide these for this test, since `estimateFeeSaga` will bail almost instantly, right?

Reply:
yeah it's not used on this test, but I started writing this test as a failing test for when we don't early bail on 0 balance and left it as is.